lib/teleterm/apiserver/apiserver.go

```diff
@@ -92,3 +111,58 @@ type APIServer struct {
 	// grpc is an instance of grpc server
 	grpcServer *grpc.Server
 }
+
+func getGrpcCredentials(cfg Config) (grpc.ServerOption, error) {
+	uri, err := utils.ParseAddr(cfg.HostAddr)
+
+	if err != nil {
+		return nil, trace.BadParameter("invalid host address: %s", cfg.HostAddr)
+	}
+
+	if uri.Network() != "unix" {
+		keyPair, err := generateKeyPair(cfg.CertsDir)
+		if err != nil {
+			return nil, trace.Wrap(err)
+		}
+
+		return grpc.Creds(keyPair), nil
+	}
+
+	return grpc.Creds(nil), nil
+}
+
+func generateKeyPair(certsDir string) (credentials.TransportCredentials, error) {
+	cert, err := utils.GenerateSelfSignedCert([]string{"localhost"})
+	if err != nil {
+		return nil, trace.Wrap(err, "failed to generate a certificate")
+	}
+
+	err = os.WriteFile(filepath.Join(certsDir, "tsh_server.crt"), cert.Cert, 0600)
```

Comment:
`"tsh_server.crt"` and `"client.crt"` should be constants with a little bit of documentation saying that Connect is expecting the exact same filenames.

Reply:
Done